### PR TITLE
Renamed Go module

### PR DESCRIPTION
**Pull request by RedbackThomson**
For future-proofing and use with other projects, I renamed the operator module name to match the Github repository URL. I did not modify the name of the smlogs module - which has its own go.mod file. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
-	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
+	golang.org/x/crypto v0.0.0-20190820162420-60c769a6c586 // indirect
+	golang.org/x/net v0.0.0-20191004110552-13f9640d40b9
+	golang.org/x/oauth2 v0.0.0-20190604053449-0f29369cfe45 // indirect
+	golang.org/x/time v0.0.0-20190308202827-9d24e82272b4 // indirect
+	google.golang.org/appengine v1.5.0 // indirect
+	k8s.io/apimachinery v0.0.0-20190816221834-a9f1d8a9c101
+	k8s.io/cli-runtime v0.0.0-20190918224932-e56234cc6b3d // indirect
+	k8s.io/client-go v11.0.1-0.20190918222721-c0e3722d5cf0+incompatible
+	k8s.io/utils v0.0.0-20191114184206-e782cd3c129f // indirect
```
Comment (gautamkmr) [inline go.mod]: It picked up automatically ?

File: smlogs-kubectl-plugin/go.mod [hunk 2]
```diff
@@ -16,7 +16,7 @@ require (
 )
 
 replace (
-	go.amzn.com/sagemaker/sagemaker-k8s-operator => ../
+	github.com/aws/amazon-sagemaker-operator-for-k8s => ../
```
[the remark this answers is not in the document]
Comment (RedbackThomson) [reply smlogs-kubectl-plugin/go.mod]: I wonder if we do that then will we need to commit changes in the operator before we can use them in smlogs? I don't think the smlogs module will know to override any changes with the local operator code.

File: go.mod (hunk 1 above)
Comment (RedbackThomson) [reply go.mod]: Correct, I did not manually override any modules.

File: smlogs-kubectl-plugin/go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (cadedaniel) [reply smlogs-kubectl-plugin/go.mod]: A little bit of chicken and egg..

I am not well versed in dependency resolution in the go runtime; you are probably right though.

**Comment by RedbackThomson** [comment]
@cadedaniel @gautamkmr The only 2 references left to the infamous `go.amzn.com` domain name are both in smlogs. Otherwise I have tested the module from my personal fork and everything worked.

**Comment by cadedaniel** [comment]
Awesome!

**Pull request merged**